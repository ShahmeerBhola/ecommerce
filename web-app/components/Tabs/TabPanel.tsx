import React from 'react';

type Props = {
  active: boolean;
};

const TabPanel: React.FunctionComponent<Props> = ({ active, children }) => (
  <div className={`bg-gray-600 p-5 shadow-2xl ${active ? 'block' : 'hidden'}`}>{children}</div>
);

export default TabPanel;
