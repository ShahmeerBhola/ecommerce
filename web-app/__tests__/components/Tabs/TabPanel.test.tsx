import React from 'react';
import renderer from 'react-test-renderer';
import TabPanel from '../../../components/Tabs/TabPanel';

describe('TabPanel component', () => {
  test('it renders properly when active is true', () => {
    const tree = renderer
      .create(
        <TabPanel active>
          <p>Tab Panel</p>
        </TabPanel>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when active is false', () => {
    const tree = renderer
      .create(
        <TabPanel active={false}>
          <p>Tab Panel</p>
        </TabPanel>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
