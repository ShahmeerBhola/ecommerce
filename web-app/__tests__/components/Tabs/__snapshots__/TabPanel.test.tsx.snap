// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanel component it renders properly when active is false 1`] = `
<div
  className="bg-gray-600 p-5 shadow-2xl hidden"
>
  <p>
    Tab Panel
  </p>
</div>
`;

exports[`TabPanel component it renders properly when active is true 1`] = `
<div
  className="bg-gray-600 p-5 shadow-2xl block"
>
  <p>
    Tab Panel
  </p>
</div>
`;
